### DISCUSSION TITLE
Why is `typeof((int A, int B))` allowed while `typeof(string?)` is not?

### DISCUSSION THREAD

**Discussion by jnm2** [dotnet/csharplang#4975] [Q&A] [answered]
`typeof(string?)` and `typeof(dynamic)` are compiler errors. Isn't the reason for this that the syntax contains extra information that can't be represented in the resulting System.Type instance, over and above `typeof(string)` and `typeof(object)`?

If so, what about the extra information in `typeof((int A, int B))` that can't be represented in the resulting System.Type instance? I would have expected the only options to be `typeof((int, int))` and `typeof(ValueTuple<int, int>)`.

[(SharpLab)](https://sharplab.io/#v2:C4LgbgpgTglgZgTwD4GIB2BXANlghgIywgAII0CiBYAKBoH1iBeY4BABwgHs4AKAAQCMABgD8ASgDcNekxbsuvACYJyAWxgBjSdOoNmrDtx48YaYMQCCAGmKnzAITHbqQA==)

**Comment by RikkiGibson** [comment, marked as answer]
See *Warn when "attribute-dependent" types are used in 'typeof'* dotnet/roslyn#54804
